src/boot/efi/device-path-util.c

```diff
@@ -3,6 +3,67 @@
 #include "device-path-util.h"
 #include "util.h"
 
+EFI_STATUS make_multiple_file_device_path(
+                EFI_HANDLE device, const char16_t **files, EFI_DEVICE_PATH ***ret_dp) {
+        EFI_STATUS err;
+        EFI_DEVICE_PATH *cur_dp = NULL, **iterator_dp = NULL;
+        EFI_DEVICE_PATH *original_device_path = NULL;
+        size_t n_files = strv_length((const void**)files);
+
+        assert(files);
+        assert(ret_dp);
+
+        err = BS->HandleProtocol(device, MAKE_GUID_PTR(EFI_DEVICE_PATH_PROTOCOL),
+                                 (void **) &original_device_path);
+        if (err != EFI_SUCCESS)
+                return err;
+
+        EFI_DEVICE_PATH *end_node = original_device_path;
+        while (!device_path_is_end(end_node))
+                end_node = device_path_next_node(end_node);
+
+        size_t o_dp_size = (uint8_t *) end_node - (uint8_t *) original_device_path;
+
+        if (n_files == 0) {
```

Comment:
this should probably be the first check, above

Reply:
Done